nodejs/device/src/index.ts

```diff
@@ -204,25 +218,30 @@ class DeviceApp {
     ) {
       appInsights.defaultClient.trackEvent({ name: "SendingPairingRequest" });
       logger.info("Updating pairing reported props: %j", pairingRequestPatch);
-      const patchReceived = this.onReceivePairingPatch();
+      let pairingAttemptPromise: Promise<ThiefPairingProperties>;
+      [
+        pairingAttemptPromise,
+        this.cancelPairingAttempt,
```

Comment:
I don't think this is going to hurt _too_ bad, but the way you save this function and never un-set it it means that the node collector will keep at least part of the `pairingAttempt` closure in memory forever.  This kind of leak is really insidious in Python -- you keep a function pointer, and the closure for that function contains an object, and that object links to other objects, and so the function pointer keeps a bunch of objects alive for much longer than they need to.  `this.cancelPairingAttempt` only has meaning while `pairingAttemptPromiose` is executing, so they seem pretty tightly bound to each other.  It feels strange to keep them in 2 different places.